contracts/registries/ThirdPartyRegistry.sol

```diff
@@ -616,6 +612,45 @@ contract ThirdPartyRegistry is OwnableInitializable, NativeMetaTransaction {
         return thirdParties[_thirdPartyId].items[_itemId];
     }
 
+    /**
+     * @notice Consume third party slots
+     * @param _consumeSlotsParams - Data to consume slots mutilple times in a single transaction
+     */
+    function _consumeSlots(string calldata _thirdPartyId, ConsumeSlotsParam[] calldata _consumeSlotsParams) internal {
+        address sender = _msgSender();
+
+        ThirdParty storage thirdParty = thirdParties[_thirdPartyId];
+
+        _checkThirdParty(thirdParty);
+
+        for (uint256 i = 0; i < _consumeSlotsParams.length; i++) {
+            ConsumeSlotsParam memory consumeSlotParam = _consumeSlotsParams[i];
+
+            if (consumeSlotParam.qty == 0) {
```

Comment:
I will convert this into a `require`. Consuming a 0 qty slot is not worth it. WDYT?

Reply:
That would fail the entire transaction intead of just skipping param. If you are ok with that I'll change it

Reply:
Yes, we should not allow that cases, emitting an event for a 0 qty slot is useless and a bug

Reply:
But the event will not be dispatched, when the qty is 0 the loop just continues to the next iteration without computing that receipt. Despite that I will change it to a require.